packages/default/scss/checkbox/_layout.scss

```diff
@@ -121,9 +121,19 @@
     .k-checkbox-label + .k-checkbox {
         margin-left: $checkbox-label-margin-x;
     }
+    .k-checkbox + kendo-label.k-checkbox-label,
+    .k-checkbox-label + kendo-label.k-checkbox {
+        margin-left: calc( #{$checkbox-label-margin-x} - 4px );
+    }
     .k-checkbox-label > .k-checkbox {
         margin-right: $checkbox-label-margin-x;
     }
+    kendo-label.k-checkbox-label > .k-checkbox {
+        margin-right: 0;
+    }
+    kendo-label.k-checkbox-label > .k-label {
+        margin-right: $checkbox-label-margin-x;
+    }
```

Comment:
You're assuming the checkbox is after the label, which might not be the case. Remove the styles or rework it to handle both cases.

```suggestion
```